### PR TITLE
Add additional C++ client documentation.

### PR DESCRIPTION
**Pull request by nomis52**
Addresses #731

### CODE REVIEW

File: doxygen/cpp-client.dox [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * The following example builds on the @ref sec_dmx_cpp_client_OLAClient_TX
+ * "OlaClient TX" example. Instead of exiting after 100 DMX frames have been
+ * sent, the program runs until the connection to olad is closed.
+ *
+ * @snippet client_disconnect.cpp Client Disconnect Example
+ *
+ * Instead of exiting, a better approach would be to try and reconnect to the
+ * olad server. Clients that do this should use a ola::BackOffPolicy to avoid
+ * spnning in a loop trying to reconnect.
```
Comment (peternewman) [inline doxygen/cpp-client.dox]: SPaG spinning

File: doxygen/cpp-client.dox [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * Sometimes it can be difficult to integrate OLA's @ref event_driven
+ * "Event Driven" programming model with the main event loop of your program.
+ * The  OlaClient is not thread safe, so a workaround is to run a separate
+ * thread for the OlaClient.
+ *
+ * We can use ola::io::SelectServer::Execute to run a callback on a
+ * SelectServer running in a different thread. This allows us to schedule the
+ * calls to OlaClient on the thread where the SelectServer is running.
+ *
```
Comment (peternewman) [inline doxygen/cpp-client.dox]: Remove the extra blank line.

File: doxygen/examples/client_disconnect.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  std::cerr << "Connection to olad was closed" << endl;
+  ss->Terminate();  // terminate the program.
+}
+
+bool SendData(ola::client::OlaClientWrapper *wrapper) {
+  static unsigned int universe = 1;
+  static unsigned int i = 0;
+  ola::DmxBuffer buffer;
+  buffer.Blackout();
+  buffer.SetChannel(0, i);
```
Comment (peternewman) [inline doxygen/examples/client_disconnect.cpp]: Would it not be helpful if the data here incremented?

**Comment by peternewman** [comment]
LGTM, just minor comments.

File: doxygen/cpp-client.dox (hunk 1 above)
Comment (nomis52) [inline doxygen/cpp-client.dox]: Fixed.

File: doxygen/cpp-client.dox (hunk 2 above)
Comment (nomis52) [inline doxygen/cpp-client.dox]: Done

File: doxygen/examples/client_disconnect.cpp (hunk 3 above)
Comment (nomis52) [inline doxygen/examples/client_disconnect.cpp]: Done

**Pull request merged**